app/core/src/main/java/com/teamg/tourdeshot/core/mapper/utils/ProductCategoryMapperUtils.java

```diff
@@ -0,0 +1,26 @@
+package com.teamg.tourdeshot.core.mapper.utils;
+
+
+import com.teamg.tourdeshot.core.model.ProductCategory;
+import com.teamg.tourdeshot.core.repository.ProductCategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategoryMapperUtils {
+
+    private final ProductCategoryRepository productCategoryRepository;
+
+    @Autowired
+    public ProductCategoryMapperUtils(ProductCategoryRepository productCategoryRepository) {
+        this.productCategoryRepository = productCategoryRepository;
+    }
+
+    public boolean hasChild(ProductCategory productCategory) {
+
+        if(productCategoryRepository.findAllByParentId(productCategory.getId()).isEmpty()) {
```

Comment:
Here you should call repository method which checks if exists category with given parentId.